python/nav/web/ipam/prefix_tree.py

```diff
@@ -406,9 +405,9 @@ def accept(prefix):
         return False
 
     heap = PrefixHeap(initial_children)
-    filtered = [prefix for prefix in prefixes if accept(prefix)]
-    nodes = [PrefixNode(prefix, sort_fn=sort_fn) for prefix in filtered]
-    for node in sorted(nodes, reverse=False):
```

Comment:
Actually, reintroducing `sorted` here seems to fix the problem with differing output between the two versions, but still keeps the improved performance.